Assets/Scripts/Item/ItemPickup.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+
+/// <summary>
+///attached to the player and when the player collides with an item, do stuff
+///each item prefab has a boxCollider set to trigger, it then will call the OnTriggerEnter2D method once collided
+///and then we can get the collided item, then we can get the itemDetails using the itemCode
+/// </summary>
+/// 
+public class ItemPickup : MonoBehaviour
+{
+
+    private void OnTriggerEnter2D(Collider2D collision)
```

Comment:
Another approach to this would be to make a serializeable itemDetails field in the Item script and assign each Item prefab an ItemDetails. This would involve more drag and drop but i can see it being more safe.